src/expression.d

```diff
@@ -6513,48 +6513,56 @@ public:
             if (fd.treq)
                 printf("  treq = %s\n", fd.treq.toChars());
         }
+        if (type)
```

Comment:
In my first implementation of "template lambda in everywhere", `FuncExp.semantic` could be called twice.
If `td` is not NULL in the first `semantic` call, `type` is set to `Type::tvoid`. And then from `cast.c` `FuncExp.semantic` was called again, to instantiated `td` with the required parameter types which were picked from the "cast to" type.

But today, the post-semantic process is handled in `FuncExp.matchType`, and the `type == Type.tvoid` check is not used anymore. That's why the check can be removed now.